### PR TITLE
Make petition form thank you url carry over existing query params 

### PR DESCRIPTION
**Pull request by mmmavis**
Related PRs/issues: #11205

---

### To QA:

1. Go to any campaign page with petition form. e.g.,https://foundation-s-11205-utm--zdqrrg.herokuapp.com/campaigns/single-page/
2. Add some query params to the URL and hit enter
3. Fill out and submit the form
4. See if the thank you page URL carries over the query params you added in step 2
5. Click on the "Yes, I'll chip in" button and see if all query params are retained as well

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py [hunk 1]
```diff
@@ -49,6 +55,15 @@ def cta(context, page):
             except ObjectDoesNotExist:
                 pass
 
+    if cta["cta_type"] == "petition":
+        source_url = context["request"].build_absolute_uri()
+
+        cta["lang"] = context["request"].LANGUAGE_CODE
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py:61]: these are all `petition` specific context

File: network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py [hunk 2]
```diff
@@ -23,16 +24,21 @@ def mini_site_horizontal_nav(context, page):
...
-        "show_formassembly_thank_you": context["request"].GET.get("thank_you") == "true",
-        "csp_nonce": context["request"].csp_nonce,
-    }
+
+    def generate_thank_you_url(url):
+        """
+        Generate a thank you page URL by grabbing the current url and appending thank_you=true
+        """
+
+        parsed_url = urlparse(url)
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py:33]: Is there a simpler implementation for this? I was thinking to just test on the existence of `?` and append `thank_you=true` accordingly lol

File: network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py [hunk 3]
```diff
@@ -23,16 +24,21 @@ def mini_site_horizontal_nav(context, page):
...
-        "cta": None,
-        "cta_type": None,
-        "lang": context["request"].LANGUAGE_CODE,
-        "source_url": context["request"].build_absolute_uri(),
-        "thank_you_url": context["request"].build_absolute_uri(context["request"].path + "?thank_you=true"),
-        "show_formassembly_thank_you": context["request"].GET.get("thank_you") == "true",
-        "csp_nonce": context["request"].csp_nonce,
-    }
+
+    def generate_thank_you_url(url):
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py:28]: Is there a simpler implementation for this helper function `generate_thank_you_url`? I was thinking to just test on the existence of `?` and append `thank_you=true` accordingly lol

Comment (jhonatan-lopes) [reply network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py:28]: The `parsed_url` seems fine, I just don't like the `"&".join`. We can get the query params as a dictionary by using:

```python
from urlib.parse import parse_qs
query_params = parse_qs(parsed_url.query)
```

Then we can append `thank_you=True` and rebuild the url. Putting it all together, I would probably re-write it as:

```python
parsed_url = urlparse(url)
query_params = parse_qs(parsed_url.query)
query_params["thank_you"] = True
query_string = urlencode(query_params)
thank_you_url = urlunparse(parsed_url._replace(query=query_string))
```

**Review by jhonatan-lopes — COMMENTED** [review]
Hi @mmmavis, I was testing here but from the code I'm not sure which parameters are allowed to pass through. Is it any parameter? I've tested it with `abc=123&def=456` and those disappeared after the form was submitted...

File: network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py (hunk 3 above)
Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py:28]: Thanks for the guidance, @jhonatan-lopes ! I didn't really like using `.join` either but wasn't able to look up for a smarter implementation lol.

Note that I ended up using `parse_qsl` as the format it returns for the key-value pairs works better here 😁 .

**Review by jhonatan-lopes — APPROVED** [review]
LGTM :+1:

**Pull request merged**